### PR TITLE
[TAS-347] ✨ Allow entering like1 address to claim nft

### PR DESCRIPTION
**Pull request by williamchong**
image

### CODE REVIEW

File: src/pages/nft/claim/index.vue [hunk 1]
```diff
@@ -171,6 +169,15 @@ export default {
       );
     },
   },
+  watch: {
+    getAddress() {
+      this.claimingAddressInput = this.getAddress;
+    },
+    loginAddress() {
+      this.claimingAddressInput = this.loginAddress;
+      this.onEnterClaimingAddress();
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:178]: Should separate the logTrackEvent from this function?

File: src/pages/nft/claim/index.vue [hunk 2]
```diff
@@ -1,81 +1,76 @@
...
-            @click="claim"
-          />
+          <template v-if="walletIsLoggingIn">
+            <ProgressIndicator />
+            <Label class="text-medium-gray w-full mt-[4px]" align="center" preset="p6"
+              :text="$t('auth_required_view_hint_label_loading')" />
+          </template>
+          <template v-else>
+            <div class="flex flex-col justify-center w-full px-[12px] mt-[12px]">
+              <Label preset="p6" class="text-medium-gray mt-[12px] mb-[6px]" :text="$t('nft_claim_enter_address_label')" />
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:23]:
```suggestion
              <Label
                class="text-medium-gray mt-[12px] mb-[6px]"
                preset="p6"
                :text="$t('nft_claim_enter_address_label')"
                align="center"
              />
```

File: src/pages/nft/claim/index.vue [hunk 3]
```diff
@@ -1,81 +1,76 @@
...
-            preset="secondary"
-            @click="claim"
-          />
+          <template v-if="walletIsLoggingIn">
+            <ProgressIndicator />
+            <Label class="text-medium-gray w-full mt-[4px]" align="center" preset="p6"
+              :text="$t('auth_required_view_hint_label_loading')" />
+          </template>
+          <template v-else>
+            <div class="flex flex-col justify-center w-full px-[12px] mt-[12px]">
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:22]:
```suggestion
            <div class="flex flex-col items-center w-full px-[12px] mt-[12px]">
```

File: src/pages/nft/claim/index.vue [hunk 4]
```diff
@@ -1,81 +1,76 @@
...
-          v-else
-          :text="$t('nft_claim_claimed_view_class_button')"
-          preset="tertiary"
-          @click="handleClickViewClass"
-        />
-      </template>
+      <template v-else>
+        <Label class="mb-[16px]" :text="text" align="center" />
+        <template v-if="state === 'INITIAL'">
+          <div class="flex flex-col justify-center w-full max-w-[680px] px-[12px] mt-[-12px]">
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:51]:
```suggestion
          <div class="flex flex-col items-center w-full max-w-[680px] px-[12px] mt-[-12px]">
```

File: src/pages/nft/claim/index.vue [hunk 5]
```diff
@@ -1,81 +1,76 @@
...
+            <Label preset="p6" class="text-medium-gray mt-[12px] mb-[6px]" :text="$t('nft_collect_modal_leave_message')" />
+            <div class="flex w-full py-[10px] px-[16px] gap-[12px] bg-shade-gray rounded-[12px]">
+              <IconMessage class="text-dark-gray" />
+              <input v-model="collectorMessage" type="input"
+                class="w-full bg-transparent border-0 focus-visible:outline-none"
+                :placeholder="$t('nft_collect_modal_leave_message_to_name', { name: creatorDisplayName })"
+                @input.once="onInputCollectorMessage">
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:58]:
```suggestion
              <input
                v-model="collectorMessage"
                class="w-full bg-transparent border-0 focus-visible:outline-none"
                :placeholder="$t('nft_collect_modal_leave_message_to_name', { name: creatorDisplayName })"
                type="input"
                @input.once="onInputCollectorMessage"
              >
```

File: src/pages/nft/claim/index.vue [hunk 6]
```diff
@@ -1,81 +1,76 @@
...
+            </div>
+            <ButtonV2 class="self-center mt-[24px]" :text="$t('nft_claim_claim')" preset="secondary" @click="claim" />
+          </div>
+        </template>
+        <ProgressIndicator v-if="state === 'CLAIMING'" class="self-center" />
+        <template v-else-if="state === 'CLAIMED'">
+          <ButtonV2 v-if="nftId" :text="$t('nft_claim_claimed_view_button')" preset="tertiary" @click="handleClickView" />
+          <ButtonV2 v-else :text="$t('nft_claim_claimed_view_class_button')" preset="tertiary"
+            @click="handleClickViewClass" />
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:67]:
```suggestion
          <ButtonV2
            v-else
            :text="$t('nft_claim_claimed_view_class_button')"
            preset="tertiary"
            @click="handleClickViewClass"
          />
```

File: src/pages/nft/claim/index.vue [hunk 7]
```diff
@@ -1,81 +1,76 @@
...
 
-      <ButtonV2
-        v-else-if="state === 'ERROR'"
-        :text="$t('nft_claim_claimed_retry_button')"
-        preset="outline"
-        @click="handleClickRetry"
-      />
-    </AuthRequiredView>
+        <ButtonV2 v-else-if="state === 'ERROR'" :text="$t('nft_claim_claimed_retry_button')" preset="outline"
+          @click="handleClickRetry" />
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:71]:
```suggestion
        <ButtonV2
          v-else-if="state === 'ERROR'"
          :text="$t('nft_claim_claimed_retry_button')"
          preset="outline"
          @click="handleClickRetry"
        />
```

File: src/pages/nft/claim/index.vue [hunk 8]
```diff
@@ -1,81 +1,76 @@
...
+              :text="$t('auth_required_view_hint_label_loading')" />
+          </template>
+          <template v-else>
+            <div class="flex flex-col justify-center w-full px-[12px] mt-[12px]">
+              <Label preset="p6" class="text-medium-gray mt-[12px] mb-[6px]" :text="$t('nft_claim_enter_address_label')" />
+              <div class="flex justify-center w-full gap-[12px]">
+                <div class="flex w-full py-[10px] px-[16px] gap-[12px] bg-shade-gray rounded-[12px] w-[300px]">
+                  <input v-model="claimingAddressInput" type="input"
+                    class="w-full bg-transparent border-0 focus-visible:outline-none"
+                    :placeholder="$t('nft_claim_enter_address_placeholder')"
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:28]:
```suggestion
                  <input
                    v-model="claimingAddressInput"
                    class="w-full bg-transparent border-0 focus-visible:outline-none"
                    :placeholder="$t('nft_claim_enter_address_placeholder')"
                    type="input"
```

File: src/pages/nft/claim/index.vue [hunk 9]
```diff
@@ -1,81 +1,76 @@
...
-      <template v-if="state === 'INITIAL'">
+    <div class="relative flex flex-col justify-center items-center w-full max-w-[962px] mx-auto mt-[4px] px-[1rem] py-[1.5rem]">
+      <NFTWidgetBaseCard class="flex justify-center items-center max-w-[400px] mb-[16px]">
+        <NuxtLink :to="localeLocation({ name: 'nft-class-classId', params: { classId } })" target="_blank">
+          <NFTWidgetContentPreview :class="[
+            'transition-shadow',
+            'cursor-pointer',
+            'min-h-[300px]',
+            'w-full',
+          ]" :title="NFTName" :description="NFTDescription" :img-src="NFTImageUrl" @click="handleClickViewDetails" />
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:11]:
```suggestion
          <NFTWidgetContentPreview
            :class="[
              'transition-shadow',
              'cursor-pointer',
              'min-h-[300px]',
              'w-full',
            ]"
            :title="NFTName"
            :description="NFTDescription"
            :img-src="NFTImageUrl"
            @click="handleClickViewDetails"
          />
```

**Review by nwingt — CHANGES_REQUESTED** [review]
The indentation is a bit weird 

Also, this PR removed the objective of https://github.com/likecoin/liker-land/pull/1291

File: src/pages/nft/claim/index.vue [hunk 10]
```diff
@@ -1,81 +1,76 @@
...
+        </NuxtLink>
+      </NFTWidgetBaseCard>
+      <template v-if="!claimingAddress">
         <div class="flex flex-col justify-center w-full max-w-[680px] px-[12px] mt-[-12px]">
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:15]:
```suggestion
        <div class="flex flex-col items-center w-full max-w-[680px] px-[12px] mt-[-12px]">
```

File: src/pages/nft/claim/index.vue (hunk 3 above)
Comment (nwingt) [inline src/pages/nft/claim/index.vue:22]: I don't think is it necessary to use this `div` as we have similar classes in parent `div` in L15, just moved the `mt-[12px]` to the parent

Parent
image

This `div`
image

Comment (williamchong) [reply src/pages/nft/claim/index.vue:22]: this div also breaks the claim NFT button to next line though

File: src/pages/nft/claim/index.vue (hunk 1 above)
Comment (williamchong) [reply src/pages/nft/claim/index.vue]: Updated to directly assign `claimingAddress`

**Review by nwingt — APPROVED** [review]

File: src/pages/nft/claim/index.vue [hunk 11]
```diff
@@ -1,82 +1,109 @@
...
+            preset="secondary"
+            :is-disabled="!claimingAddressInput || !isValidAddress(claimingAddressInput)"
+            @click="onEnterClaimingAddress"
+          />
+        </template>
+      </template>
+      <template v-else>
+        <Label v-if="text" class="mb-[16px]" :text="text" align="center" />
+        <template v-if="state === 'INITIAL'">
+          <Label preset="p6" class="text-medium-gray mt-[12px] mb-[6px]" :text="$t('nft_collect_modal_leave_message')" />
```
Comment (nwingt) [inline src/pages/nft/claim/index.vue:69]:
```suggestion
          <Label preset="p6" class="text-medium-gray mb-[6px]" :text="$t('nft_collect_modal_leave_message')" />
```

**Review by nwingt — APPROVED** [review]

**Pull request merged**